### ISSUE TITLE
Issue with SharedFile.getInputStream()

### ISSUE THREAD

**Issue by reisin1986** [swaechter/smbjwrapper#1]
Hi! I'm testing your wrapper and all examples work well. Really great wrapper! 
But the last example (Download to an output stream) is not working:
```
[main] INFO com.hierynomus.smbj.session.Session - Logging off session 4398650490985 from host vm.test.com
[main] INFO com.hierynomus.smbj.connection.Connection - Closed connection to vm.test.com
[Packet Reader for vm.test.com] INFO com.hierynomus.smbj.transport.PacketReader - Thread[Packet Reader for vm.test.com,5,main] stopped.
com.hierynomus.mssmb2.SMBApiException: STATUS_ACCESS_DENIED(3221225506/3221225506): Create failed for test.txt
	at com.hierynomus.smbj.share.Share.receive(Share.java:342)
	at com.hierynomus.smbj.share.Share.sendReceive(Share.java:322)
	at com.hierynomus.smbj.share.Share.createFile(Share.java:135)
	at com.hierynomus.smbj.share.DiskShare.createFileAndResolve(DiskShare.java:79)
	at com.hierynomus.smbj.share.DiskShare.open(DiskShare.java:69)
	at com.hierynomus.smbj.share.DiskShare.openFile(DiskShare.java:144)
	at ch.swaechter.smbjwrapper.SharedFile.getInputStream(SharedFile.java:53)
	at Main.main(Main.java:25)
```

Problem is here:                 
InputStream inputStream = sharedFile.getInputStream();

Any ideas?

Note: I use smbj version 0.8.0

**Comment by swaechter** [comment]
Hey thank you very much. It started as a proof of concept to replace JCIFS at the company I work for, so feel free to request new features ;)

But back to the issue: Can you provide a minimal working example that does reflect your problem? If so, I can test your code tomorrow and provide a fix + bump the smbj version.

Also important:

- What SMB server are you using (Linux Samba, Windows Active Directory or local Share or other vendors like for example a Synology NAS)?
- Are you connected as guest or authenticated as indicated in the message (User/password or User/Domain and password)?

All in all I just need enough information to replicate the problem :)

Best,
Simon

**Comment by reisin1986** [comment]
Hi, Simon! I'm glad to hear you answer!
I'm trying to migrate from jcifs to smbj in my project. Smbj is great, but it doesn't has a good documentation and, sometimes, it's hard to understand some things there. 
With your wrapper I've made main part of functionality - there is issue only with downloading from share.

I made some investigation and found, that this error appears only if remote sharedFile has attribute "ReadOnly", If I clear this attribut manually (clear checkbox in file attributes) - all works well!
But, I think - there should be way to copy readonly files? As  I understand - "readonly" attribute should not affect functionality for downloading?

My testing environment:
Local: Windows 10(not domain - default workgroup)
Remote: Windows 10 with share opened for all users.

Default smb dialect is 3.1.1 (but with support all above 2.0)

Sincerely,
Alex

**Comment by swaechter** [comment]
Hey Alex

Yeah, it's a great low-level library with a lot of features, but it also took me some time to wrap the whole functionality I was looking for.

I was able to reproduce your problem and just fixed it with the next version 0.0.5 (0.0.4 was opening files with all generic rights like read/write - not only read). The 0.0.5 version should appear in the Maven repository within the next hour.

Can you try the 0.0.5 version and confirm if it's also working for you?

Best,
Simon
